integrations/GA/browser.js

```diff
@@ -1,122 +1,986 @@
+        enhancedEcommerceTrackProductAction(
+          rudderElement,
+          "detail",
+          data,
+          this.inputs
+        );
+        pushEnhancedEcommerce(rudderElement, this.inputs);
+      } else if (event === "Product Clicked" && this.enhancedEcommerce) {
+        var props = rudderElement.message.properties;
+        var data = {};
+        this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+          rudderElement,
+          this.enhancedEcommerceLoaded
+        );
+
+        if (props.list) data.list = props.list;
+        enhancedEcommerceTrackProductAction(
+          rudderElement,
+          "click",
+          data,
+          this.inputs
+        );
+        pushEnhancedEcommerce(rudderElement, this.inputs);
+      } else if (event === "Promotion Viewed" && this.enhancedEcommerce) {
+        var props = rudderElement.message.properties;
+
+        this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+          rudderElement,
+          this.enhancedEcommerceLoaded
+        );
+        ga("ec:addPromo", {
+          id: props.promotionId || props.id,
+          name: props.name,
+          creative: props.creative,
+          position: props.position,
+        });
+        pushEnhancedEcommerce(rudderElement, this.inputs);
+      } else if (event === "Promotion Clicked" && this.enhancedEcommerce) {
+        var props = rudderElement.message.properties;
+        this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+          rudderElement,
+          this.enhancedEcommerceLoaded
+        );
+
+        ga("ec:addPromo", {
+          id: props.promotionId || props.id,
+          name: props.name,
+          creative: props.creative,
+          position: props.position,
+        });
+        ga("ec:setAction", "promo_click", {});
+        pushEnhancedEcommerce(rudderElement, this.inputs);
+      } else if (event === "Product List Viewed" && this.enhancedEcommerce) {
+        var props = rudderElement.message.properties;
+        var products = props.products;
+
+        this.enhancedEcommerceLoaded = loadEnhancedEcommerce(
+          rudderElement,
+          this.enhancedEcommerceLoaded
+        );
+
+        each(products, function (product) {
+          var item = { properties: product };
+          if (
+            !(item.properties.product_id || item.properties.sku) &&
+            !item.properties.name
+          )
+            return;
+          var impressionObj = {
+            id: item.properties.productId || item.properties.sku,
+            name: item.properties.name,
+            category: item.properties.category || props.category,
+            list: props.list_id || props.category || "products",
+            brand: item.properties.band,
+            variant: item.properties.variant,
+            price: item.properties.price,
+            position: getProductPosition(item, products),
+          };
+          impressionObj = extend(
+            impressionObj,
+            metrics(
```

Comment:
whereever adding metrics to payload/properties/product, have a seperate method that adds them to the base object.